torch/fx/graph.py

```diff
@@ -101,10 +113,78 @@ def create_node(self, op: str, target: Target,
         self._mark_uses(args)
         self._mark_uses(kwargs)
         n = Node(self, name if name is not None else self._name(target), op, target, args, kwargs)
-        assert self._nodes[-1].op == 'return'
-        self._nodes.insert(-1, n)
+        self.insert_node_before(n=n, before=self._insert_point)
         return n
 
+    def insert_node_before(self, n : Node, before : Node) -> Node:
+        """
+        Insert node `n` directly before `before` in the topological ordering of nodes
+        """
+        # TODO: computationally inefficient
+        before_idx = self._nodes.index(before)
```

Comment:
In the common case of creating a graph, this hits the worst case N^2 behavior every time. While it is probably acceptable for having the mutation operators be inefficient to start, making _every_ graph creation N^2 is not.

Reply:
How is this N^2? Inserting one-before-the-end should still be N, right? I.e. every iteration we would only need to move that one element (amortized for resizing)

Anyway, this implementation is going to totally change after i rebase past making `output` a node